src/Illuminate/Log/LogManager.php

```diff
@@ -159,13 +159,17 @@ protected function parseTap($tap)
     /**
      * Create an emergency log handler to avoid white screens of death.
      *
-     * @return \Psr\Log\LoggerInterface
+     * @return \Illuminate\Log\Logger|\Psr\Log\LoggerInterface
      */
     protected function createEmergencyLogger()
     {
-        return new Logger(new Monolog('laravel', $this->prepareHandlers([new StreamHandler(
-                $this->app->storagePath().'/logs/laravel.log', $this->level(['level' => 'debug'])
-        )])), $this->app['events']);
+        $config = $this->configurationFor('emergency');
+
+        return new Logger(
+            new Monolog('laravel', $this->prepareHandlers([
+                new StreamHandler($config['path'], $this->level(['level' => 'debug'])),
```

Comment:
How is it ensured it still falls back to the "old way" if that particular config is not present?

(there is not `emergency` in the default laravel install ATM)

Reply:
Check the referenced pull request for laravel/laravel above your comment.

Reply:
https://github.com/laravel/laravel/pull/5179

Reply:
I'm blind 🤦‍♂ , makes sense!

**However**: this needs to be upgrade guide and or have simply fallback here to easy the pain.

Example:
- this is the emergency logger
- it relies now on a config
- if you upgrade to a Laravel version and did not adapt
- then in case an of emergency you got totally nothing back

But basically this becomes a very critical configuration every really should have then.

At the very least, `$config['path']` should have a `?? storage_path('logs/laravel.log')` or something _in here_.

Reply:
You're very right! I'll make that change now!

Thanks for pointing out the issue!

Reply:
https://github.com/Stokoe0990/framework/blob/configurable-emergency-logger/src/Illuminate/Log/LogManager.php#L164

I've updated that now.

Reply:
Thanks, LGTM 👍 

I guess `configurationFor()` simply returns empty array/null when the config isn't present but _no_ error thrown?

Reply:
https://github.com/Stokoe0990/framework/blob/configurable-emergency-logger/src/Illuminate/Log/LogManager.php#L427

```php
    /**
     * Get the log connection configuration.
     *
     * @param  string  $name
     * @return array
     */
    protected function configurationFor($name)
    {
        return $this->app['config']["logging.channels.{$name}"];
    }
```

> I guess configurationFor() simply returns empty array/null when the config isn't present but no error thrown?

Yes :)